### PR TITLE
refactor: extract autonomy.test_tools into aea-test-autonomy plugin

### LINKED ISSUE #1158: interdependency of skills and their tests
There are tests for skill where we import classes from other skill while we should not. I have not systematically assessed the situation, but will list some of my findings here to illustrate.

https://github.com/valory-xyz/open-autonomy/blob/3ea8ff010d2b6ed55059ff43ee916a1371aff143/tests/test_skills/test_price_estimation_abci/test_rounds.py#L30-L33
https://github.com/valory-xyz/open-autonomy/blob/3ea8ff010d2b6ed55059ff43ee916a1371aff143/tests/test_skills/test_safe_deployment_abci/test_rounds.py#L26-L29
https://github.com/valory-xyz/open-autonomy/blob/3ea8ff010d2b6ed55059ff43ee916a1371aff143/tests/test_skills/test_transaction_settlement_abci/test_rounds.py#L37

the same thing goes for reuse of parts from tests. If not from `test_abstract_round.py` These should reside in some `base.py` or `helpers.py` script, e.g.
https://github.com/valory-xyz/open-autonomy/blob/3ea8ff010d2b6ed55059ff43ee916a1371aff143/tests/test_skills/test_safe_deployment_abci/test_rounds.py#L42-L48

Suggestion:

- test that concrete skills do not have any imports from other concrete skills
- a test for the same in the tests modules of concrete tests

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR solves various issues:
- some packages depended on `autonomy` (not `aea`) - this is not allowed and would not work given current deployment flow! To fix this, this PR extracts the `autonomy.test_tools` folder as an `open-aea` compatible plugin which can be referenced in the packages directly.
- fixes dependencies in tests of packages in anticipation of tests moving into packages (also thanks to #1260) - Fixes #1158

## Fixes

- #1158

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [x] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: autonomy/deploy/base.py [hunk 1]
```diff
@@ -43,7 +43,6 @@
 ABCI_HOST = "abci{}"
 TENDERMINT_NODE = "http://node{}:26657"
 TENDERMINT_COM = "http://node{}:8080"
-LOCALHOST = "localhost"
```
Comment (DavidMinarsch) [inline autonomy/deploy/base.py:46]: Not used

File: packages/valory/skills/liquidity_rebalancing_abci/test_tools/integration.py [hunk 2]
```diff
@@ -22,8 +22,8 @@
 from pathlib import Path
 from typing import Any
 
-from autonomy.test_tools.fixture_helpers import HardHatAMMBaseTest
```
Comment (DavidMinarsch) [inline packages/valory/skills/liquidity_rebalancing_abci/test_tools/integration.py:25]: This was a big (undetected) problem. Deploying these as a service would not have worked in the new deployment flow.

File: packages/valory/skills/liquidity_rebalancing_abci/test_tools/integration.py [hunk 3]
```diff
@@ -22,8 +22,8 @@
...
-from autonomy.test_tools.helpers.contracts import get_register_contract
+from aea_test_autonomy.fixture_helpers import HardHatAMMBaseTest
```
Comment (DavidMinarsch) [inline packages/valory/skills/liquidity_rebalancing_abci/test_tools/integration.py:25]: This is now referenced as a skill dependency in the `skill.yaml`

File: packages/valory/skills/transaction_settlement_abci/test_tools/integration.py [hunk 4]
```diff
@@ -30,11 +30,10 @@
 from aea.crypto.base import Crypto
 from aea.crypto.registries import make_crypto, make_ledger_api
 from aea_ledger_ethereum import EthereumApi
+from aea_test_autonomy.fixture_helpers import HardHatAMMBaseTest
+from aea_test_autonomy.helpers.contracts import get_register_contract
 from web3.types import Nonce, Wei
 
-from autonomy.test_tools.fixture_helpers import HardHatAMMBaseTest
```
Comment (DavidMinarsch) [inline packages/valory/skills/transaction_settlement_abci/test_tools/integration.py:35]: same as above

File: plugins/aea-test-autonomy/aea_test_autonomy/configurations.py [hunk 5]
```diff
@@ -36,6 +36,13 @@ def get_key(key_path: Path) -> str:
 LOCALHOST = "localhost"
 HTTP_LOCALHOST = f"http://{LOCALHOST}"
 
+DEFAULT_IMAGE_VERSION = "latest"
+TENDERMINT_IMAGE_VERSION = os.environ.get(
+    "TENDERMINT_IMAGE_VERSION", DEFAULT_IMAGE_VERSION
+)
+TENDERMINT_IMAGE_NAME = os.environ.get(
+    "TENDERMINT_IMAGE_NAME", "valory/open-autonomy-tendermint"
+)
```
Comment (DavidMinarsch) [inline plugins/aea-test-autonomy/aea_test_autonomy/configurations.py:45]: Duplication from `autonomy`. Will add a test to ensure they stay synced!

File: plugins/aea-test-autonomy/setup.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    name="open-aea-test-autonomy",
+    version="0.2.1",
+    author="Valory AG",
+    license="Apache-2.0",
+    description="Plugin containing test tools for open-autonomy packages.",
+    packages=find_packages(
+        where=".", include=["aea_test_autonomy", "aea_test_autonomy.*"]
+    ),
+    entry_points={},
+    install_requires=["open-aea>=1.0.0, <2.0.0"],
```
Comment (DavidMinarsch) [inline plugins/aea-test-autonomy/setup.py:36]: TBD ... didn't check yet

File: tests/conftest.py [hunk 7]
```diff
@@ -95,6 +95,12 @@
 }
 
 
+skip_docker_tests = pytest.mark.skipif(
```
Comment (DavidMinarsch) [inline tests/conftest.py:98]: Moved out of test tools/plugin, feels not appropriate there!

File: tests/test_protocols/test_abci/test_abci_spec.py [hunk 8]
```diff
@@ -28,6 +28,7 @@
     _get_sub_types_of_compositional_types,
 )
 
+# TOFIX - deps not allowed -> move file to connection
```
Comment (DavidMinarsch) [inline tests/test_protocols/test_abci/test_abci_spec.py:31]: TODO

File: tests/test_skills/test_abstract_round_abci/test_common.py [hunk 9]
```diff
@@ -324,6 +324,7 @@ def test_clean_up(
         assert self.behaviour.context.randomness_api._retries_attempted == 0
 
 
+# TOFIX: depends on transaction_settlement_abci, so should not be here in this form
```
Comment (DavidMinarsch) [inline tests/test_skills/test_abstract_round_abci/test_common.py:327]: TODO

File: tests/test_skills/test_counter/test_models.py [hunk 10]
```diff
@@ -20,6 +20,7 @@
 """Test the models.py module of the skill."""
 from unittest.mock import MagicMock
 
+# TOFIX - wrong dep
```
Comment (DavidMinarsch) [inline tests/test_skills/test_counter/test_models.py:23]: TODO

File: tests/test_skills/test_liquidity_provision_abci/test_behaviours_integration.py [hunk 11]
```diff
@@ -65,6 +64,8 @@
 from packages.valory.skills.transaction_settlement_abci.payload_tools import (
     hash_payload_to_hex,
 )
+
+# TOFIX - perhaps ok, needs checking
```
Comment (DavidMinarsch) [inline tests/test_skills/test_liquidity_provision_abci/test_behaviours_integration.py:68]: TODO

File: tests/test_skills/test_liquidity_rebalancing_abci/test_rounds.py [hunk 12]
```diff
@@ -36,12 +36,16 @@
     SynchronizedData,
     TransactionHashBaseRound,
 )
+
+# TOFIX - dep not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_liquidity_rebalancing_abci/test_rounds.py:40]: TODO

File: tests/test_skills/test_oracle_deployment_abci/test_rounds.py [hunk 13]
```diff
@@ -47,10 +47,14 @@
     SynchronizedData as OracleDeploymentSynchronizedSata,
 )
 from packages.valory.skills.oracle_deployment_abci.rounds import ValidateOracleRound
+
+# TOFIX - not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_oracle_deployment_abci/test_rounds.py:51]: TODO

File: tests/test_skills/test_price_estimation_abci/test_payloads.py [hunk 14]
```diff
@@ -27,6 +27,8 @@
 from packages.valory.skills.price_estimation_abci.payloads import (
     TransactionType as PETransactionType,
 )
+
+# TOFIX - not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_price_estimation_abci/test_payloads.py:31]: TODO

File: tests/test_skills/test_safe_deployment_abci/test_behaviours.py [hunk 15]
```diff
@@ -46,6 +46,8 @@
     BaseRandomnessBehaviourTest,
     BaseSelectKeeperBehaviourTest,
 )
+
+# TOFIX - dep not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_safe_deployment_abci/test_behaviours.py:50]: TODO

File: tests/test_skills/test_safe_deployment_abci/test_rounds.py [hunk 16]
```diff
@@ -23,6 +23,8 @@
 from typing import Dict, FrozenSet, Optional
 
 from packages.valory.skills.abstract_round_abci.base import AbciAppDB, MAX_INT_256
+
+# TOFIX - dep not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_safe_deployment_abci/test_rounds.py:27]: TODO

File: tests/test_skills/test_transaction_settlement_abci/test_behaviours.py [hunk 17]
```diff
@@ -99,6 +99,8 @@
     BaseRandomnessBehaviourTest,
     BaseSelectKeeperBehaviourTest,
 )
+
+# TOFIX - deps not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_transaction_settlement_abci/test_behaviours.py:103]: TODO

File: tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py [hunk 18]
```diff
@@ -53,9 +52,13 @@
     HandlersType,
     IntegrationBaseCase,
 )
+
+# TOFIX - deps not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py:56]: TODO

File: tests/test_skills/test_transaction_settlement_abci/test_rounds.py [hunk 19]
```diff
@@ -34,6 +34,8 @@
     MAX_INT_256,
     TransactionNotValidError,
 )
+
+# TOFIX - deps not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_transaction_settlement_abci/test_rounds.py:38]: TODO

File: tox.ini [hunk 20]
```diff
@@ -337,16 +339,17 @@ deps =
...
 skip_install = True
 deps =
     mypy==0.910
-commands = mypy autonomy packages/valory --disallow-untyped-defs
-           mypy tests --disallow-untyped-defs
-           mypy scripts --disallow-untyped-defs
+commands = 
+        python -m pip install --no-deps file://{toxinidir}/plugins/aea-test-autonomy
+        mypy autonomy packages/valory plugins/aea-test-autonomy/aea_test_autonomy tests scripts --disallow-untyped-defs
+        mypy plugins/aea-test-autonomy/tests --disallow-untyped-defs
```
Comment (DavidMinarsch) [inline tox.ini:352]: One cannot have two modules of same name `tests` in the same analysis

File: .github/workflows/main_workflow.yml [hunk 21]
```diff
@@ -66,8 +66,8 @@ jobs:
       run: tox -e check-copyright
     - name: Check docs
       run: tox -e check-api-docs
-    - name: Check doc links and IPFS hashes
-      run: tox -e check-doc-links-hashes
+    # - name: Check doc links and IPFS hashes
```
Comment (DavidMinarsch) [inline .github/workflows/main_workflow.yml:69]: temp

File: tests/test_skills/test_abstract_round_abci/test_common.py [hunk 22]
```diff
@@ -42,9 +42,6 @@
 from packages.valory.skills.abstract_round_abci.test_tools.base import (
     FSMBehaviourBaseCase,
 )
-from packages.valory.skills.transaction_settlement_abci.payload_tools import (
```
Comment (DavidMinarsch) [inline tests/test_skills/test_abstract_round_abci/test_common.py:45]: Not required

File: tests/test_skills/test_liquidity_rebalancing_abci/test_rounds.py [hunk 23]
```diff
@@ -36,7 +37,6 @@
     SynchronizedData,
     TransactionHashBaseRound,
 )
-from packages.valory.skills.price_estimation_abci.payloads import TransactionHashPayload
```
Comment (DavidMinarsch) [inline tests/test_skills/test_liquidity_rebalancing_abci/test_rounds.py:39]: wrong import

File: tests/test_skills/test_oracle_deployment_abci/test_rounds.py [hunk 24]
```diff
@@ -114,30 +107,10 @@ def get_safe_contract_address() -> str:
...
-) -> Dict[str, EstimatePayload]:
-    """participant_to_estimate"""
-    return {
-        participant: EstimatePayload(sender=participant, estimate=1.0)
-        for participant in participants
-    }
-
-
-def get_participant_to_tx_hash(
-    participants: FrozenSet[str], hash_: Optional[str] = "tx_hash"
```
Comment (DavidMinarsch) [inline tests/test_skills/test_oracle_deployment_abci/test_rounds.py:136]: not used

File: tests/test_skills/test_oracle_deployment_abci/test_rounds.py [hunk 25]
```diff
@@ -47,14 +48,6 @@
...
 )
 from packages.valory.skills.oracle_deployment_abci.rounds import ValidateOracleRound
-from packages.valory.skills.price_estimation_abci.payloads import (
-    EstimatePayload,
-    TransactionHashPayload,
-)
-from packages.valory.skills.transaction_settlement_abci.payload_tools import (
-    VerificationStatus,
-)
-from packages.valory.skills.transaction_settlement_abci.payloads import ValidatePayload
```
Comment (DavidMinarsch) [inline tests/test_skills/test_oracle_deployment_abci/test_rounds.py:57]: wrong imports

File: tests/test_skills/test_price_estimation_abci/test_payloads.py [hunk 26]
```diff
@@ -27,34 +27,6 @@
 from packages.valory.skills.price_estimation_abci.payloads import (
     TransactionType as PETransactionType,
 )
-from packages.valory.skills.safe_deployment_abci.payloads import (
```
Comment (DavidMinarsch) [inline tests/test_skills/test_price_estimation_abci/test_payloads.py:30]: Moved

File: tests/test_skills/test_safe_deployment_abci/test_payloads.py [hunk 27]
```diff
@@ -0,0 +1,62 @@
...
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Test the payloads.py module of the skill."""
+
+from packages.valory.skills.safe_deployment_abci.payloads import (
```
Comment (DavidMinarsch) [inline tests/test_skills/test_safe_deployment_abci/test_payloads.py:22]: Moved here from other file

File: tests/test_skills/test_safe_deployment_abci/test_rounds.py [hunk 28]
```diff
@@ -37,12 +38,14 @@
     SynchronizedData as SafeDeploymentSynchronizedSata,
 )
 from packages.valory.skills.safe_deployment_abci.rounds import ValidateSafeRound
-from packages.valory.skills.transaction_settlement_abci.payloads import ValidatePayload
 
+# TOFIX - dep not allowed
```
Comment (DavidMinarsch) [inline tests/test_skills/test_safe_deployment_abci/test_rounds.py:42]: @angrybayblade please finish

File: tests/test_protocols/test_abci/test_abci_spec.py (hunk 8 above)
Comment (DavidMinarsch) [reply tests/test_protocols/test_abci/test_abci_spec.py:31]: @angrybayblade

File: tests/test_skills/test_transaction_settlement_abci/test_behaviours.py (hunk 17 above)
Comment (DavidMinarsch) [reply tests/test_skills/test_transaction_settlement_abci/test_behaviours.py:103]: @angrybayblade

File: tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py (hunk 18 above)
Comment (DavidMinarsch) [reply tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py:56]: @angrybayblade

File: tests/test_skills/test_transaction_settlement_abci/test_rounds.py (hunk 19 above)
Comment (DavidMinarsch) [reply tests/test_skills/test_transaction_settlement_abci/test_rounds.py:38]: @angrybayblade

File: docs/autonomy.md [hunk 29]
```diff
@@ -75,7 +75,11 @@ These keys can be used for local deployments if you're using the default hardhat
 
 ```bash
 # fetch a service
+<<<<<<< HEAD
```
Comment (DavidMinarsch) [inline docs/autonomy.md:78]: fix

File: docs/price_oracle_intro.md [hunk 30]
```diff
@@ -41,7 +41,11 @@ you have followed the [setup instructions](quick_start.md#setup). As a result yo
 
 2. Use the CLI to download the `valory/oracle_hardhat` service.
     ```bash
+<<<<<<< HEAD
```
Comment (DavidMinarsch) [inline docs/price_oracle_intro.md:44]: fix

File: tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py (hunk 18 above)
Comment (DavidMinarsch) [reply tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py:56]: Fixed

File: plugins/aea-test-autonomy/aea_test_autonomy/helpers/base.py [hunk 31]
```diff
@@ -76,14 +73,6 @@ def try_send(gen: Generator, obj: Any = None) -> None:
         gen.send(obj)
 
 
-def make_round_class(name: str, bases: Tuple = (AbstractRound,)) -> Type[AbstractRound]:
```
Comment (DavidMinarsch) [inline plugins/aea-test-autonomy/aea_test_autonomy/helpers/base.py:79]: Wrong place! Cannot have package dependencies

File: tests/test_skills/test_abstract_round_abci/test_abci_app_chain.py [hunk 32]
```diff
@@ -19,16 +19,28 @@
...
 )
-from packages.valory.skills.abstract_round_abci.base import AbciApp, DegenerateRound
+from packages.valory.skills.abstract_round_abci.base import (
+    AbciApp,
+    AbstractRound,
+    DegenerateRound,
+)
+
+
+def make_round_class(name: str, bases: Tuple = (AbstractRound,)) -> Type[AbstractRound]:
```
Comment (DavidMinarsch) [inline tests/test_skills/test_abstract_round_abci/test_abci_app_chain.py:38]: Moved here

File: tests/test_skills/test_oracle_deployment_abci/test_rounds.py (hunk 25 above)
Comment (Karrenbelt) [reply tests/test_skills/test_oracle_deployment_abci/test_rounds.py:57]: takes care of #1158

File: plugins/aea-test-autonomy/aea_test_autonomy/docker/tendermint.py [hunk 33]
```diff
@@ -139,6 +141,7 @@ def __create_flask_tendermint_image(self) -> None:
         try:
             self._client.images.get(self.tag)
         except ImageNotFound:
+            # TOFIX - remove.
```
Comment (Karrenbelt) [inline plugins/aea-test-autonomy/aea_test_autonomy/docker/tendermint.py:144]: this remains to be done still, is it a separate issue or simply forgotten?

Comment (DavidMinarsch) [reply plugins/aea-test-autonomy/aea_test_autonomy/docker/tendermint.py:144]: Separate issue

**Review by Karrenbelt — APPROVED** [review]
Failures unit / integration tests are among the usual suspects, same goes for e2e tests which I saw fail in earlier PR as well. Don't know why `TestRegistrationStartUpFourAgents` and `TestRegistrationStartUpFourAgentsCatchUp` fail though, they didn't use to.

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**